pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go

```diff
@@ -237,7 +239,76 @@ func TestTransitiveClosure(t *testing.T) {
 			},
 			want: wantGraph{ids: []string{"a", "b", "c"}},
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "don't traverse error nodes",
+			cloud: mockCloud,
+			graph: func() *rgraph.Builder {
+				g := rgraph.NewBuilder()
+				addNodeToGraph(g, "a", []string{"b", "c"})
+				addNode("b", []string{"d", "e"}, nodeIsInError)
+				addNode("d", []string{"a"})
+				addNode("e", []string{"b"})
+				return g
+			},
+			want:    wantGraph{ids: []string{"a", "b", "c"}},
+			wantErr: true,
+		},
+		{
+			name:  "don't traverse error nodes - child in error",
```

Comment:
I think we need to define the behavior in the case where there is an error encountered. Do we expect that the traverse will continue and we get as much of the graph as we can OR will the traversal be canceled and the result be just whatever was fetched up around the time we detected the error and tried to stop?

If we can add this to the doc strings as well and also change the documented name of the test cases, this is will be more obvious to maintainers.

Reply:
yes, you are right there is no point checking the output on the error.